packages/core/src/envelope.ts

```diff
@@ -91,6 +91,14 @@ export function createEventEnvelope(
   // have temporarily added, etc. Even if we don't happen to be using it at some point in the future, let's not get rid
   // of this `delete`, lest we miss putting it back in the next time the property is in use.)
   delete event.sdkProcessingMetadata;
+  try {
+    // @ts-expect-error - for bundle size we try/catch the access to this property
+    delete event.contexts.trace.data._sentry_span_name_set_by_user;
```

Comment:
m: Instead of doing this here, can we handle this in otel? We have a function `removeSentryAttributes` there already that gets rid of "internal" attributes, IMHO we can just add to this? :)

Reply:
unfortunately I'm afraid we have to do this in core because people can call `updateSpanName` in browser SDKs as well.

Reply:
Hmm, I still think it would be nicer to not do this in the envelope sending here, but in the place where `data` is generated. 

So for node, in `removeSentryAttributes`, and for browser/core, maybe in `_convertSpanToTransaction` of `SentrySpan`? This appears cleaner to me 🤔

Reply:
Ok, sounds fine to me, too. I'll make the change.

Reply:
[24e8320](https://github.com/getsentry/sentry-javascript/pull/14291/commits/24e83207dceaecb4a8858db99792f649c609dcdf)